packages/blocks/src/api/registration.js

```diff
@@ -232,6 +232,20 @@ export function registerBlockType( blockNameOrMetadata, settings ) {
 		return;
 	}
 
+	if (
+		settings?.parent &&
+		Array.isArray( settings.parent ) &&
+		settings.parent.length === 1 &&
+		settings.parent[ 0 ] === name
+	) {
```

Comment:
```suggestion
	if ( settings?.parent?.[ 0 ] === name && settings.parent.length === 1 ) {
```

Nit: I don't have a strong opinion, but this might be simpler.

Reply:
Sure, I'll update the code and add test case